### DISCUSSION TITLE
Add files larger than INT32_MAX to zip archive

### DISCUSSION THREAD

**Discussion by FedericoGnesotto** [nmoinvaz/minizip#697] [Q&A] [answered]
Hi all,
I hope someone can help me with this. I have been trying to add a file which is around 2.5Gb large (larger than INT32_MAX) to a zip archive via the `zipWriteInFileInZip(..)` function. I got an error (UNZ_PARAMERROR) so I looked at the source code and found out that an error is returned if the size of the file is larger than INT32_MAX (mz_compat.c, line 580).
My question is then: how can I add a file which is larger than INT32_MAX to an archive? do I have to use the new API? If yes, what is the best way to do that?

Thanks a lot in advance for your help,

Federico

**Comment by Nekto89** [comment, marked as answer]
I'm calling zipWriteInFileInZip multiple times.
```
while (i_size != 0) {
  // minizip-ng has limit in zipWriteInFileInZip
  constexpr auto max_chunk_size = static_cast<std::size_t>(std::numeric_limits<std::int32_t>::max() - 1);
  const auto write_size = std::min(max_chunk_size, i_size);
  i_size -= write_size;
  if (zipWriteInFileInZip(mp_impl->mh_archive, ip_buffer, static_cast<std::uint32_t>(write_size)) < 0)
    return false;
  ip_buffer += write_size;
}
return true;
```

**Reply by FedericoGnesotto** [reply]
Thanks! I thought I tried this but maybe I had some mistake in there, will give it another try!
